### PR TITLE
Add ability to hide (soft delete) chats to inference backend

### PR DESCRIPTION
**Pull request by olliestanley**

### CODE REVIEW

File: inference/server/oasst_inference_server/routes/chats.py [hunk 1]
```diff
@@ -270,3 +271,12 @@ async def handle_update_title(
...
         logger.exception("Error when updating chat title")
         return fastapi.Response(status_code=500)
+
+
+@router.put("/{chat_id}/hide")
+async def hide_chat(
+    chat_id: str,
+    ucr: UserChatRepository = Depends(deps.create_user_chat_repository),
+):
+    await ucr.hide_chat(chat_id)
```
Comment (AbdBarho) [inline inference/server/oasst_inference_server/routes/chats.py:281]: can we make this a toggle? i.e. hide / unhide, either as a boolean parameter (default hidden) or simply inverting whatever is the database.

Motivation: users would expect the ability to make stuff visible again, unlike delete, which is gone permanently.

**Comment by notmd** [comment]
Is this for soft delete chat or sharing chat?

**Comment by olliestanley** [comment]
> Is this for soft delete chat or sharing chat?

Soft delete

File: inference/server/oasst_inference_server/models/chat.py [hunk 2]
```diff
@@ -74,12 +74,15 @@ class DbChat(SQLModel, table=True):
 
     messages: list[DbMessage] = Relationship(back_populates="chat")
 
+    hidden: bool = Field(False, index=True)
```
[the remark this answers is not in the document]
Comment (olliestanley) [reply inference/server/oasst_inference_server/models/chat.py:77]: I did it with speeding up queries in mind, but in hindsight given we already filter by `user_id` first whenever we are listing non-hidden chats I guess the speedup is minimal. I will remove this index

**Comment by yk** [comment]
shouldn't we default `include_hidden` to False?

**Review by yk — APPROVED** [review]

**Pull request merged**